### PR TITLE
Add recommendation for anti-spam

### PR DESCRIPTION
**Pull request by jboland88**
#### Changes proposed in this Pull Request:

* This PR adds a new recommendation that provides some information about managing spam comments that shows up when a post receives 5+ comments.

#### Jetpack product discussion

pbNhbs-1iw-p2

#### Does this pull request change what data or activity we track or use?

This PR adds a new resource recommendation step and uses existing analytics events introduced in #23600 to record interactions with the recommendation on the individual step screen and on the summary screen. Analytics events that can be recorded for this new step are:
```
// Individual Step:
jetpack_recommended_resource_learn_more_click
jetpack_recommended_resource_read_click
jetpack_recommended_resource_skip_click

// Summary Screen:
jetpack_recommendations_summary_learn_more_click
```

#### Testing instructions:

To reset the assistant state while testing, you can use the following commands ( you can run these with `jetpack docker` if using the Docker development environment )
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_conditional`
* `wp jetpack options delete recommendations_step`

Note that if your test site has the Akismet plugin enabled, you will not receive the new recommendation as outlined in the test plan below.

---

* Checkout this branch on your Jetpack test site.
* Visit the Jetpack dashboard. Note that the "Recommendations" tab in the dashboard navigation does not show any badge indicating that there is a new recommendation.
Screen Shot 2022-03-21 at 2 27 58 PM

* Now, visit the front end of the site and go to an individual post.
* Add sample comments to the post until the post has 5 or more comments.
* Navigate back to the Jetpack Dashboard. You should see a red badge with a "1" next to the Recommendations tab in the navigation on the Jetpack dashboard - indicating a new recommendation is available.
Screen Shot 2022-03-11 at 4 00 23 PM

* Click on the "Recommendations" tab in the navigation - you should now see a recommendation step that provides some information about spam comments. There should be a green "New" badge that shows on this step in the upper-left corner. The CTA button should link out to this article: https://jetpack.com/2021/04/29/why-spam-comments-exist-and-how-to-stop-them/
Screen Shot 2022-03-30 at 9 49 08 AM

* Click on "Read Later"  to proceed to the summary screen. On the summary screen, you should see the a new "Resources" section with the "Spam Management" recommendation listed. A green "New" badge should show next to "Spam Management" in the list. The "Read More" button should link to the same article mentioned in the previous step.
Screen Shot 2022-03-30 at 9 48 54 AM

* If you reload the page on the summary screen, the green "New" badge next to Site Security in the summary resources list and the red "1" indicator next to "Recommendations" in the dashboard navigation  should no longer show.
* Automated unit tests should be passing
* Visuals should match this reference: G151IYl8Lm4f0NjgLpR943-fi-2132%3A39570

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (travisw) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:219]: When adding a comment on the front end I'm getting a fatal error that `is_plugin_active()` isn't defined. 🤔 
Is it possible that function hasn't loaded yet at the point when this runs?
Screen Shot 2022-04-01 at 1 36 37 PM

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (travisw) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:283]: Noting that this function is also being defined in https://github.com/Automattic/jetpack/pull/23678.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (jboland88) [reply projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:219]: Thanks for catching that! I think `is_plugin_active` only loads in the admin context 🤔 I wonder why this didn't fail for me! Leaving this codex page here for my future reference :) 
https://developer.wordpress.org/reference/functions/is_plugin_active/

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (jboland88) [reply projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:283]: 😅 Thanks! I think I may wait until the eventual merge conflict pops up to resolve as opposed to updating and rebasing from the base branch.

**Review by monsieur-z — CHANGES_REQUESTED** [review]
Everything works well, and code looks good.

Just one thing: when a user clicks on _Learn how to block spam_, the wizard should probably move to the summary screen. Otherwise, when a user opens the article and switches tab back to the Assistant, they're presented with these 2 options: read it again, or read it later. It's not clear that they need to click _Read Later_ to move on.

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (jeherve) [inline projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:237]: Maybe we could rely on `Plugins_Installer::is_plugin_active()` here?

**Comment by monsieur-z** [comment]
> Great suggestion @monsieur-z! I've updated the behavior accordingly.

Perfect 👍

File: projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php
Comment (jboland88) [reply projects/plugins/jetpack/_inc/lib/class-jetpack-recommendations.php:237]: Ah, good idea :)

**Review by monsieur-z — APPROVED** [review]
Everything works as expected, nice work 👍

**Review by kraftbj — APPROVED** [review]

**Pull request merged**
